core/src/main/java/org/commonjava/indy/core/expire/ScheduleKey.java

```diff
@@ -101,4 +109,31 @@ public boolean exists()
     {
         return this.storeKey != null && this.type != null;
     }
+
+    @Override
+    public void writeExternal( ObjectOutput out )
+            throws IOException
+    {
+        out.writeObject( storeKey.getName() );
+        out.writeObject( storeKey.getType().name() );
+        out.writeObject( storeKey.getPackageType() );
+        out.writeObject( type );
+        out.writeObject( name );
+    }
+
+    @Override
+    public void readExternal( ObjectInput in )
+            throws IOException, ClassNotFoundException
+    {
+        final String storeKeyName = (String) in.readObject();
+        final StoreType storeType = StoreType.get( (String) in.readObject() );
+        final String packageType = (String)in.readObject();
+        storeKey = new StoreKey( packageType, storeType, storeKeyName );
+
+        final String typeStr = (String) in.readObject();
```

Comment:
It seems not need this. If write(null) readobject also returns null.

Reply:
Some protection here. Not sure if the read value is null or "".